rskj-core/src/test/java/co/rsk/core/CallContractTest.java

```diff
@@ -82,7 +82,8 @@ private static ProgramResult callContract(World world, RskAddress receiveAddress
                     blockFactory,
                     new ProgramInvokeFactoryImpl(),
                     new PrecompiledContracts(config, bridgeSupportFactory, world.getBlockTxSignatureCache()),
-                    world.getBlockTxSignatureCache()
+                    world.getBlockTxSignatureCache(),
+                    null
```

Comment:
Is this really necessary? If this field can be nullable, we could just create a constructor without it and call the old one with null no?

Something like this:

```
    public TransactionExecutorFactory(
            RskSystemProperties config,
            BlockStore blockStore,
            ReceiptStore receiptStore,
            BlockFactory blockFactory,
            ProgramInvokeFactory programInvokeFactory,
            PrecompiledContracts precompiledContracts,
            BlockTxSignatureCache blockTxSignatureCache) {
        this(config, blockStore, receiptStore, blockFactory, programInvokeFactory, precompiledContracts, blockTxSignatureCache, null);
    }
```

Reply:
You are right, I'll update it.

Reply:
In this [commit](https://github.com/rsksmart/rskj/pull/2276/commits/1b367e3e10d961f54bd67936c4833d3c364c724b) I reverted this and similar changes so this is nor longer an issue.

Reply:
I saw this change, I will make some comments there on the class created.